homeassistant/components/weather/__init__.py

```diff
@@ -1050,6 +1050,8 @@ async def async_update_listeners(
         self, forecast_types: Iterable[Literal["daily", "hourly", "twice_daily"]] | None
     ) -> None:
         """Push updated forecast to all listeners."""
+        if not hasattr(self, "_forecast_listeners"):
+            return
```

Comment:
Check is necessary if entity is updated before added as `self._forecast_listeners` doesn't exist yet.